src/hist.jl

```diff
@@ -213,3 +215,150 @@ fit{N,W}(::Type{Histogram}, vs::NTuple{N,AbstractVector}, wv::WeightVec{W}, edge
     append!(Histogram(edges,W,closed), vs, wv)
 fit{N}(::Type{Histogram},vs::NTuple{N,AbstractVector}, wv::WeightVec; closed::Symbol=:right, nbins=sturges(length(vs[1]))) =
     fit(Histogram, vs, wv, histrange(vs,nbins,closed); closed=closed)
+
+
+# Get a suitable high-precision type for the norm of a histogram.
+@generated function norm_type{T, N, E}(h::Histogram{T, N, E})
+    args = [:( eltype(edges[$d]) ) for d = 1:N]
+    quote
+        edges = h.edges
+        norm_type(promote_type(T, $(args...)))
+    end
+end
+
+norm_type{T<:Integer}(::Type{T}) = promote_type(T, Int64)
+norm_type{T<:AbstractFloat}(::Type{T}) = promote_type(T, Float64)
+
+
+"""
+    norm(h::Histogram)
+
+Calculate the norm of histogram `h` as the absolute value of its integral.
+"""
+@generated function norm{T, N, E}(h::Histogram{T, N, E})
+    quote
+        edges = h.edges
+        weights = h.weights
+        S = norm_type(h)
+        v_0 = 1
+        s_0 = zero(S)
+        @inbounds @nloops(
+            $N, i, weights,
+            d -> begin
+                v_{$N-d+1} = v_{$N-d} * (edges[d][i_d + 1] - edges[d][i_d])
+                s_{$N-d+1} = zero(S)
+            end,
+            d -> begin
+                s_{$N-d} += s_{$N-d+1}
+            end,
+            begin
+                $(Symbol("s_$(N)")) += (@nref $N weights i) * $(Symbol("v_$N"))
+            end
+        )
+        norm(s_0)
+    end
+end
+
+
+function _float_deepcopy(x)
+    y = float(x)
+    y === x? deepcopy(y) : y
```

Comment:
Won't save any code though, since I also need `_float_deepcopy` for `Base.float(h::Histogram)` (this should come in handy at times) - which I defined not to copy if not necessary, to be consistent with the behavior of float for arrays.

Also, float(A) may have higher performance than float.(A) - not sure.